docs/support/tutorials/gis/eo_guide.md

```diff
@@ -39,221 +54,166 @@ What to consider when chosing data:
+[**QGIS**](../../../apps/qgis.md) - is a very widely used GUI for working with spatial data, it has also limited multispectral image processing capabilities.  GUI with batch processing possibility and Python interface. Used for example for visualization, map algebra and other raster processing. Many plug-ins available, for EO data processing, check out the [QGIS Semi-automatic classification plugin](https://fromgistors.blogspot.com/p/semi-automatic-classification-plugin.html).
 
+[**R**](../../../apps/r-env-for-gis.md) - Puhti R installation includes a lot of geospatial packages, includeing serveral useful for EO data processing.
 
-[**QGIS**](../../../apps/qgis.md)
+[**Sen2Cor**](../../../apps/sen2cor.md) - a command-line tool for Sentinel-2 Level 2A product generation and formatting.
 
-* General Geographic Information System (GIS) with limited multispectral image processing capabilities
-* GUI with batch processing possibility and Python interface
-* Used for example for visualization, map algebra and other raster processing
-* Many plug-ins available, for EO data processing, check out the [QGIS Semi-automatic classification plugin](https://fromgistors.blogspot.com/p/semi-automatic-classification-plugin.html).
+[**Sen2mosaic**](../../../apps/sen2cor.md) - a command-line tool to download, preprocess and mosaic Sentinel-2 data.
 
-[**R**](../../../apps/r-env-for-gis.md)
+[**SNAP**](../../../apps/snap.md) - ESA Sentinel Application Platform. Tool for processing of Sentinel data (+ support for other data sources). GUI, CLI (Graph Processing Tool, GPT) and Python interfaces. [SNAP GPT example for Puhti](https://github.com/csc-training/geocomputing/tree/master/snap).
 
-* All R packages are provided in `r-env` module, including many that can be used for EO data processing, such as terra, CAST, raster, spacetime.
-
-[**Sen2Cor**](../../../apps/sen2cor.md)
+If you need further applications, you can ask CSC to install them for you. 
+ 
+### Machine Learning with EO data
 
-* A command-line tool for Sentinel-2 Level 2A product generation and formatting.
+One example of the advanced usage of EO data is for machine learning. If you are interested in the topic, you can find a lot of examples from [CSC machine learning with spatial data course materials]( https://github.com/csc-training/GeoML). For practical guidelines, see also [CSC machine learning guide](../ml-guide.md)
 
-[**Sen2mosaic**](../../../apps/sen2cor.md)
 
-* A command-line tool to download, preprocess and mosaic Sentinel-2 data.
+## Alternative processing services
+Below is a list of alternative EO processing services that might be useful, when a lot of data is required and downloading it all to CSC might not be feasible. All of them include the main open datasets: Sentinel, Landsat, MODIS etc.
```

Comment:
```suggestion
Below is a list of alternative EO platforms that might be useful, when a lot of data is required and downloading it all to CSC might not be feasible. 
```

Reply:
I would not call them all platforms.. Also why not include the main datasets?

Reply:
"main datasets": I don't think it adds anything here. Not everyone is looking for those datasets.
EO processing service ok.